src/LanguageServer/Impl/LanguageServer.Configuration.cs

```diff
@@ -67,11 +68,50 @@ public sealed partial class LanguageServer {
                 HandleUserConfiguredPathsChanges(userConfiguredPaths);
                 HandlePathWatchChanges(GetSetting(analysis, "watchSearchPaths", true));
                 HandleDiagnosticsChanges(pythonSection, settings);
+                HandleCodeActionsChanges(pythonSection);
 
                 _server.DidChangeConfiguration(new DidChangeConfigurationParams { settings = settings }, cancellationToken);
             }
         }
 
+        private void HandleCodeActionsChanges(JToken pythonSection) {
+            var refactoring = new Dictionary<string, object>();
+            var quickFix = new Dictionary<string, object>();
+
+            var refactoringToken = pythonSection["refactoring"];
+            var quickFixToken = pythonSection["quickfix"];
+
+            // +1 is for last "." after prefix
+            AppendToMap(refactoringToken, refactoringToken?.Path.Length + 1 ?? 0, refactoring);
+            AppendToMap(quickFixToken, quickFixToken?.Path.Length + 1 ?? 0, quickFix);
+
+            var codeActionSettings = new CodeActionSettings(refactoring, quickFix);
+            _server.HandleCodeActionsChange(codeActionSettings);
+
+            void AppendToMap(JToken setting, int prefixLength, Dictionary<string, object> map) {
+                if (setting == null || !setting.HasValues) {
+                    return;
+                }
+
+                foreach (var child in setting) {
+                    if (child is JValue value) {
+                        // there shouldn't be duplicates and prefix must exist.
+                        var path = child.Path;
+                        if (path.Length <= prefixLength) {
+                            // nothing to add
+                            continue;
+                        }
+
+                        // get rid of common "settings.python..." prefix
+                        map[path.Substring(prefixLength)] = value.Value;
```

Comment:
Maybe just pick stuff after the last dot? Up to you.

Reply:
I would like people to have something like

python.quickfix.addimport.prefer_import = true;
python.quickfix.addimport = true;
python.quickfix.addimport.abbreviation = { "np" : "numpy", "pd": "panda" }

and access them as

settings.GetQuickfixSettings("addimport.prefer_import", false) or
settings.GetQuickfixSettings("addimport", true) or
settings.GetQuickfixSettings("addimport.abbreviation", EmptyDictionary ) 

something like that.